src/sentry/deletions/base.py

```diff
@@ -177,12 +177,11 @@ def extend_relations_bulk(self, child_relations, obj_list):
     def chunk(self, num_shards=None, shard_id=None):
         """
         Deletes a chunk of this instance's data. Return ``True`` if there is
-        more work, or ``False`` if the entity has been removed.
+        more work, or ``False`` if all matching entities have been removed.
         """
         query_limit = self.query_limit
-        remaining = self.chunk_size
         has_more = True
-        while remaining > 0:
+        while has_more:
```

Comment:
Is `has_more` actually useful anymore? We now never break out of this loop until the queryset is empty. 

Does this break the chunking functionality? I think the intent here is that after we process a chunk we fire a new task so that we don't end up with one really long running task.

Reply:
This does sacrifice small chunks for fewer errors, and deletions that don't take months to process. The previous implementation would:

1. Start deleting an organization.
2. Delete the first 100 commits.
3. Attempt to delete all of the commit authors and fail as some of them are still reference.
4. The task would die, and would be restarted on the following day.

This cycle would repeat until all the commits in an organization were removed.

Reply:
I've pushed an update for this method. However, there is also another tight loop in `delete_children`. This loop was added in #9664 to prevent us losing our place in the relations list and having to re-emit select/delete queries. I've added a metric to see how many tasks would come from starting new tasks for each chunk, as I suspect it is quite a few.

Reply:
Ahh, I didn't realise that the next chunk would happen a day later... Our usual pattern here would be to just fire another task almost immediately, but I guess that gets lost in the abstractions of the deletion code.